src/database_connection.cpp

```diff
@@ -79,16 +80,74 @@ std::string updateStatement(UpdateReturnQuery query, std::string returning, bool
   return "";
 }
 
-std::string getFilterStatement(GenericFilter filter) {
-  if (filter.type == FilterTypes::COMPARISON) {
-    return std::string{filter.a + " BETWEEN " +
-                       filter.b + " AND " +
-                       filter.comparison};
-  } else {
-    return std::string{filter.a + filter.comparison + filter.b};
+template <typename T>
+std::string filterStatement(T filter) {
+  std::string filter_string{};
+  if constexpr(std::is_same_v<T, MultiOptionFilter>) {
+    filter_string += filter.a + " " + filter.comparison + " (";
+    std::string delim = "";
+    for (const auto& option : filter.options) {
+      filter_string += delim + option;
+      delim = ",";
+    }
+    filter_string += ")";
+  } else if constexpr(std::is_same_v<T, CompBetweenFilter>) {
+    filter_string += filter.field + " BETWEEN " +
+                  filter.a + " AND " +
+                  filter.b;
+  } else if constexpr(std::is_same_v<T, CompFilter>) {
+    filter_string += filter.a + filter.sign + filter.b;
+  }
+  return std::move(filter_string);
+}
+
+template <typename FilterA, typename FilterB>
+std::string getVariantFilterStatement(std::vector<std::variant<FilterA, FilterB>> filters) {
+  std::string filter_string{};
+  auto idx = 0;
+  auto filter_count = filters.size();
+  for (const auto& filter : filters) {
+    if (filter.index() == 0) {
+      filter_string += filterStatement(std::get<0>(filter));
+    } else {
+      filter_string += filterStatement(std::get<1>(filter));
```

Comment:
can there be more than 1 filter index?

Reply:
The filter index range is dictated by the number of template arguments being given to `std::variant` on line 105